test/integration/test_bitcoin_com_market.py

```diff
@@ -0,0 +1,450 @@
+        self.assertEqual(order_id, order_completed_event.order_id)
+        self.assertAlmostEqual(quantized_amount, order_completed_event.base_asset_amount)
+        self.assertEqual("ETH", order_completed_event.base_asset)
+        self.assertEqual("BTC", order_completed_event.quote_asset)
+        self.assertAlmostEqual(base_amount_traded, order_completed_event.base_asset_amount)
+        self.assertAlmostEqual(quote_amount_traded, order_completed_event.quote_asset_amount)
+        self.assertTrue(any([isinstance(event, BuyOrderCreatedEvent) and event.order_id == order_id
+                             for event in self.market_logger.event_log]))
+        # Reset the logs
+        self.market_logger.clear()
+
+    def test_market_sell(self):
+        trading_pair = "ETHBTC"
+        amount: Decimal = Decimal("0.0001")
+        quantized_amount: Decimal = self.market.quantize_order_amount(trading_pair, amount)
+
+        order_id = self.market.sell(trading_pair, amount, OrderType.MARKET, Decimal(0))
+        [order_completed_event] = self.run_parallel(self.market_logger.wait_for(SellOrderCompletedEvent))
+        order_completed_event: SellOrderCompletedEvent = order_completed_event
+        trade_events = [t for t in self.market_logger.event_log if isinstance(t, OrderFilledEvent)]
+        base_amount_traded = sum(t.amount for t in trade_events)
+        quote_amount_traded = sum(t.amount * t.price for t in trade_events)
+
+        self.assertTrue([evt.order_type == OrderType.LIMIT for evt in trade_events])
+        self.assertEqual(order_id, order_completed_event.order_id)
+        self.assertAlmostEqual(quantized_amount, order_completed_event.base_asset_amount)
+        self.assertEqual("ETH", order_completed_event.base_asset)
+        self.assertEqual("BTC", order_completed_event.quote_asset)
+        self.assertAlmostEqual(base_amount_traded, order_completed_event.base_asset_amount)
+        self.assertAlmostEqual(quote_amount_traded, order_completed_event.quote_asset_amount)
+        self.assertTrue(any([isinstance(event, SellOrderCreatedEvent) and event.order_id == order_id
+                             for event in self.market_logger.event_log]))
+        # Reset the logs
+        self.market_logger.clear()
+
+    def test_cancel_order(self):
+        trading_pair = "ETHBTC"
+
+        current_bid_price: Decimal = self.market.get_price(trading_pair, True)
+        amount: Decimal = Decimal("0.0001")
+        self.assertGreater(self.market.get_balance("ETH"), amount)
+
+        bid_price: Decimal = current_bid_price * Decimal("0.5")
+        quantize_bid_price: Decimal = self.market.quantize_order_price(trading_pair, bid_price)
+        quantized_amount: Decimal = self.market.quantize_order_amount(trading_pair, amount)
+
+        client_order_id = self.market.buy(trading_pair, quantized_amount, OrderType.LIMIT, quantize_bid_price)
+        self.market.cancel(trading_pair, client_order_id)
```

Comment:
The `test_cancel_order` is passing occasionally. I found out that by waiting for a `BuyOrderCreatedEvent` would solve this issue.

Simply include after the `self.market.buy` function:
`self.run_parallel(self.market_logger.wait_for(BuyOrderCreatedEvent))`

Reply:
I have addressed this issue [here](https://github.com/CoinAlpha/hummingbot/pull/1153/commits/b89ee7e1dcbcba66bd33d73910412d5176da51c6).

Thanks for the feedback!